apps/platform/src/projects/ProjectAdminController.ts

```diff
@@ -2,27 +2,55 @@ import Router from '@koa/router'
 import { extractQueryParams } from '../utilities'
 import { searchParamsSchema } from '../core/searchParams'
 import { ProjectState } from '../auth/AuthMiddleware'
+import { addAdminToProject, getProjectAdmin, pagedProjectAdmins, removeAdminFromProject } from './ProjectAdminRepository'
+import { JSONSchemaType } from 'ajv'
 import { Admin } from '../auth/Admin'
-import { addAdminToProject, pagedProjectAdmins, updateAdminProjectState } from './ProjectAdminRepository'
+import { validate } from '../core/validate'
+import { projectRoleMiddleware } from './ProjectService'
+import { ProjectAdminParams } from './ProjectAdmins'
+import { projectRoles } from './Project'
 
 const router = new Router<
     ProjectState & { admin?: Admin }
 >({
     prefix: '/admins',
 })
 
+router.use(projectRoleMiddleware('admin'))
+
 router.get('/', async ctx => {
     const params = extractQueryParams(ctx.query, searchParamsSchema)
     ctx.body = await pagedProjectAdmins(params, ctx.state.project.id)
 })
 
-router.post('/', async ctx => {
-    await addAdminToProject(ctx.state.project.id, ctx.request.body.admin_id)
-    ctx.body = true
+const projectAdminParamsSchema: JSONSchemaType<ProjectAdminParams> = {
+    $id: 'projectAdminParams',
+    type: 'object',
+    required: ['role'],
+    properties: {
+        role: {
+            type: 'string',
+            enum: projectRoles,
+        },
+    },
+}
+
+router.put('/:adminId', async ctx => {
+    const admin = await Admin.find(ctx.params.adminId)
+    if (!admin) return ctx.throw(404, 'invalid adminId')
```

Comment:
Could pull this out into a param call like we do in other controllers where we check the model (and then that covers `put`, `get`, `delete`, etc

Reply:
i would but the "put" method still works even if there is no `project_admins` record yet